presto-main/src/main/java/com/facebook/presto/operator/HttpTaskResultClient.java

```diff
@@ -69,467 +51,65 @@
-                            location,
-                            backoff.getFailureCount(),
-                            backoff.getFailureDuration().convertTo(SECONDS),
-                            backoff.getFailureRequestTimeTotal().convertTo(SECONDS));
-                    t = new PrestoException(REMOTE_BUFFER_CLOSE_FAILED, message, t);
+                if (familyForStatusCode(response.getStatusCode()) != HttpStatus.Family.SUCCESSFUL) {
+                    log.debug("Unexpected acknowledge response code: %s", response.getStatusCode());
                 }
-                handleFailure(t, resultFuture);
+                return null;
             }
-        }, pageBufferClientCallbackExecutor);
-    }
-
-    private static void checkNotHoldsLock(Object lock)
-    {
-        checkState(!Thread.holdsLock(lock), "Cannot execute this method while holding a lock");
-    }
-
-    private void handleFailure(Throwable t, HttpResponseFuture<?> expectedFuture)
-    {
-        // Can not delegate to other callback while holding a lock on this
-        checkNotHoldsLock(this);
-
-        requestsFailed.incrementAndGet();
-        requestsCompleted.incrementAndGet();
-
-        if (t instanceof PrestoException) {
-            clientCallback.clientFailed(PageBufferClient.this, t);
-        }
-
-        synchronized (PageBufferClient.this) {
-            if (future == expectedFuture) {
-                future = null;
-            }
-            lastUpdate = DateTime.now();
-        }
-        clientCallback.requestComplete(PageBufferClient.this);
-    }
-
-    @Override
-    public boolean equals(Object o)
-    {
-        if (this == o) {
-            return true;
-        }
-        if (o == null || getClass() != o.getClass()) {
-            return false;
-        }
-
-        PageBufferClient that = (PageBufferClient) o;
-
-        if (!location.equals(that.location)) {
-            return false;
-        }
-
-        return true;
+        });
     }
 
     @Override
-    public int hashCode()
+    public ListenableFuture<?> abortResults()
     {
-        return location.hashCode();
-    }
-
-    @Override
-    public String toString()
-    {
-        String state;
-        synchronized (this) {
-            if (closed) {
-                state = "CLOSED";
-            }
-            else if (future != null) {
-                state = "RUNNING";
-            }
-            else {
-                state = "QUEUED";
-            }
-        }
-        return toStringHelper(this)
-                .add("location", location)
-                .addValue(state)
-                .toString();
-    }
-
-    private static Throwable rewriteException(Throwable t)
-    {
-        if (t instanceof ResponseTooLargeException) {
-            return new PageTooLargeException();
-        }
-        return t;
+        return httpClient.executeAsync(prepareDelete().setUri(location).build(), createStatusResponseHandler());
```

Comment:
Since we don't need the returned `StatusResponse` future. Can we just transform the future to be a `ListenableFuture <Void>`?  This would avoid the `FutureCallback<Object>` in https://github.com/prestodb/presto/pull/13894/commits/00e38a39d8d24b7800cf46b4ad284cf4270371a4#diff-e3a5fa934c535b087cd4f3c2bfe24b90R369